docs/program/extend/modular-resources/_index.md

```diff
@@ -1,150 +1,528 @@
+You can see built-in Viam resource APIs in the <a href="https://github.com/viamrobotics/api" target="_blank">Viam GitHub</a>.
+{{% /alert %}}
 
 ### Models
 
-A model is an implementation of a resource type that implements all or some of the API methods of a resource type API.
-Models allow you to control any number of versions of a given resource with a consistent interface.
-This is powerful, because you only need to learn and code against one interface which remains the same for different models of the same component type.
+A *model* descrines a specific implementation of a resource that implements (speaks) its API.
+Models allow you to control different versions of resource types with a consistent interface.
+
+For example:
 
-For example, some DC motors can be controlled with GPIO, which you can interface with in different ways depending on the attached controlling hardware.
-Other DC motors are controlled with various serial protocols.
-This is simplified with Viam, as any motor model that implements the *rdk:component:motor* API can be powered with the *SetPower* method.
+- Some DC motors use just [GPIO](/components/board/#using-gpio), while other DC motors use serial protocols like [SPI bus](/components/board/#spi-bus).
+- Regardless, any motor model that implements the *rdk:component:motor* API can be powered with the `SetPower()` method.
 
-Viam represents models with *colon-delimited-triplets*.
-For example, the __rdk:builtin:gpio__ model of the __rdk:component:motor__ API provides RDK support for GPIO-controlled DC motors and the __rdk:builtin:DMC4000__ model of the same __rdk:component:motor__ API provides support for the DMC 4000 motor.
+Models are also uniquely namespaced as colon-delimited-triplets in the form of `namespace:family:name`.
+
+For example:
+
+- The `rdk:builtin:gpio` model of the `rdk:component:motor` API provides RDK support for [GPIO-controlled DC motors](/components/motor/gpio/).
+- The `rdk:builtin:DMC4000` model of the same `rdk:component:motor` API provides RDK support for the [DMC4000](/components/motor/dmc4000/) motor.
 
 A common use-case for modular resources is to create a new model using an existing Viam API.
 However, you can also create and expose new API types using modular resources.
 
-## Modular Resource Management with the RDK
+## Use a modular resource with your robot
 
-### Dependency Management
+Add a modular resource to your robot configuration in five steps:
 
-Modular resources may depend on other built-in resources or other modular resources, and vice versa.
-The Viam RDK handles dependency management.
+1. Code a module in Go or Python, using the module support libraries provided by the Python or Go [Viam SDK](/program/sdk-as-client).
+2. Compile or package the module code into an executable.
+3. Save the executable in a location your `viam-server` instance can access.
+4. Add a **module** referencing this executable to the configuration of your robot.
```

Comment:
in the app, right? Should probably specify that

Reply:
actually all of these should just also have a link to the section where there's more info